Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilder.cpp

```diff
@@ -188,15 +188,31 @@ namespace ROS2
         const auto fullSourcePath = AZ::IO::Path(request.m_watchFolder) / AZ::IO::Path(request.m_sourceFile);
 
         AZ_Info(SdfAssetBuilderName, "Parsing source file: %s", fullSourcePath.c_str());
-        auto parsedSourceFile = UrdfParser::ParseFromFile(fullSourcePath.String());
-        if (!parsedSourceFile)
+        auto parsedSdfRootOutcome = UrdfParser::ParseFromFile(fullSourcePath);
+        if (!parsedSdfRootOutcome)
         {
-            AZ_Error(SdfAssetBuilderName, false, "Failed to parse source file '%s'.", fullSourcePath.c_str());
+            AZStd::string sdfImportErrors;
+            for (const sdf::Error& sdfError : parsedSdfRootOutcome.error())
+            {
+                AZStd::string errorMessage = AZStd::string::format("ErrorCode=%d", static_cast<int32_t>(sdfError.Code()));
+                errorMessage += AZStd::string::format(", Message=%s", sdfError.Message().c_str());
+                if (sdfError.LineNumber().has_value())
+                {
+                    errorMessage += AZStd::string::format(", Line=%d", sdfError.LineNumber().value());
+                }
+                sdfImportErrors += errorMessage;
+                sdfImportErrors += '\n';
+            }
```

Comment:
This error composing block seems to be the same same as in ROS2RobotImporterEditorSystemComponent (repeated code).

One more occurrence.

Reply:
It is the same.
I'll add a function, that can convert an `sdf::Errors` type alias which is a `std::vector<sdf::Error>` into an `AZStd::string`

Reply:
The errors should be merged with `xacro` errors, e.g. for [ur5 robot](https://github.com/RobotecAI/o3de-ros2-gem-testing/blob/main/URDFTest/03_IMPORTING_URDF_1/IMPORTING_URDF1_TESTCASE.md) the list of errors is as follows:
![Screenshot from 2023-08-18 15-39-49](https://github.com/o3de/o3de-extras/assets/134940295/6f497563-f865-44dd-a96e-5716519bfb54)
while the terminal shows some meaningful outcome:
```
Warning [parser_urdf.cc:2774] Error Code 19: Msg: urdf2sdf: link[base_link] has no <inertial> block defined. Please ensure this link has a valid mass to prevent any missing links or joints in the resulting SDF model.
Error Code 19: Msg: urdf2sdf: allowing joint lumping by removing any <disableFixedJointLumping> or <preserveFixedJoint> gazebo tag on fixed parent joint[base_joint], as well as ensuring that ParserConfig::URDFPreserveFixedJoint is false, could help resolve this warning. See http://sdformat.org/tutorials?tut=sdformat_urdf_extensions for more information about this behavior.
Error Code 19: Msg: urdf2sdf: parent joint[base_joint] ignored.
Error Code 19: Msg: urdf2sdf: allowing joint lumping by removing any <disableFixedJointLumping> or <preserveFixedJoint> gazebo tag on fixed child joint[base_link-base_fixed_joint], as well as ensuring that ParserConfig::URDFPreserveFixedJoint is false, could help resolve this warning. See http://sdformat.org/tutorials?tut=sdformat_urdf_extensions for more information about this behavior.
Error Code 19: Msg: urdf2sdf: allowing joint lumping by removing any <disableFixedJointLumping> or <preserveFixedJoint> gazebo tag on fixed child joint[base_link-base_link_inertia], as well as ensuring that ParserConfig::URDFPreserveFixedJoint is false, could help resolve this warning. See http://sdformat.org/tutorials?tut=sdformat_urdf_extensions for more information about this behavior.
Error Code 19: Msg: urdf2sdf: [2] child joints ignored.
Error Code 19: Msg: urdf2sdf: [2] child links ignored.
Warning [parser_urdf.cc:2777] Error Code 19: Msg: urdf2sdf: link[base_link] is not modeled in sdf.

```